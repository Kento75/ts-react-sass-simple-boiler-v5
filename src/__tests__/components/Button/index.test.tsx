import * as React from 'react';
import { shallow } from 'enzyme';
import Button from '../../../components/Button';

test('小コンポーネントが存在すること', () => {
  const wrapper = shallow(
    <Button text="ボタンです" flag={true} action={() => console.log('test')} />
  );

  expect(wrapper.find('button').length).toBe(1);
  expect(wrapper.find('p').length).toBe(1);

  expect(wrapper.find('p').text()).toEqual('ボタンです');

  expect(wrapper).toMatchSnapshot();
});

test('pコンポーネントが表示されないこと', () => {
  const wrapper = shallow(
    <Button text="ボタンです" flag={false} action={() => console.log('test')} />
  );

  expect(wrapper.find('button').length).toBe(1);
  expect(wrapper.find('p').length).toBe(0);

  expect(wrapper).toMatchSnapshot();
});

test('イベント発火時にコールバック関数が呼び出されること', () => {
  const Spy = jest.fn();
  const wrapper = shallow(
    <Button text="ボタンです" flag={true} action={Spy} />
  );

  wrapper.find('button').simulate('click');
  expect(Spy).toHaveBeenCalledWith();

  expect(wrapper).toMatchSnapshot();
});
