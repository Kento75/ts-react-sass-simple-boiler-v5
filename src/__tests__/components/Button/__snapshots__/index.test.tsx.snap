// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pコンポーネントが表示されないこと 1`] = `
<Fragment>
  <button
    onClick={[Function]}
  >
    Button
  </button>
</Fragment>
`;

exports[`イベント発火時にコールバック関数が呼び出されること 1`] = `
<Fragment>
  <p>
    ボタンです
  </p>
  <button
    onClick={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
  >
    Button
  </button>
</Fragment>
`;

exports[`小コンポーネントが存在すること 1`] = `
<Fragment>
  <p>
    ボタンです
  </p>
  <button
    onClick={[Function]}
  >
    Button
  </button>
</Fragment>
`;
